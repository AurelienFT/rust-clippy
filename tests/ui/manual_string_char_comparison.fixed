#![warn(clippy::manual_string_char_comparison)]
#![allow(unused)]

fn main() {
    let sentence = "Hello, world!";
    ;
    sentence.split(|c: char| c == '\n' || c == 'X');
    sentence.split(|c: char| matches!(c, '\n' | 'X'));
}
