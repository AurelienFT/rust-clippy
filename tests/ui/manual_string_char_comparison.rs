#![warn(clippy::manual_string_char_comparison)]
#![allow(unused)]

fn main() {
    let sentence = "Hello, world!";
    sentence.trim_end_matches(|c: char| c == '.' || c == '!' || c == '?');
    sentence.split(|c: char| c == '\n' || c == 'X');
    sentence.split(|c: char| matches!(c, '\n' | 'X'));
}
