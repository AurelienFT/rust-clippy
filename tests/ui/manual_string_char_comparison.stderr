error: This manual char comparaison can be written more succinctly
  --> tests/ui/manual_string_char_comparison.rs:6:5
   |
LL |     sentence.trim_end_matches(|c: char| c == '.' || c == '!' || c == '?');
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try
   |
   = note: `-D clippy::manual-string-char-comparison` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::manual_string_char_comparison)]`

error: aborting due to 1 previous error

